Helper/Cart.php

```diff
@@ -2859,8 +2860,8 @@ public function getSaleRuleDiscounts($quote)
         $ruleDiscountDetails = [];
         foreach ($salesRuleIds as $salesRuleId) {
             $rule = $this->ruleRepository->getById($salesRuleId);
-            // Only collect non-zero discount / free shipping promotion
-            if (!$rule || (!array_key_exists($salesRuleId, $saleRuleDiscountsDetails) && !$rule->getSimpleFreeShipping())) {
+            // Only collect non-zero discount / free shipping promotion(with coupon code)
+            if (!$rule || (!array_key_exists($salesRuleId, $saleRuleDiscountsDetails) && $rule->getCouponType() == RuleInterface::COUPON_TYPE_NO_COUPON)) {
```

Comment:
To reduce UI changes can we do exactly the same change as we did before refactoring?
I think it was before foreach loop?

Reply:
OK, I revert the condition statement https://github.com/BoltApp/bolt-magento2/blob/b2b0a6daba5421f31fc8b98e3c2c8d2629433068/Helper/Cart.php#L2193